bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/customdatatype/datatype/CodedConceptDatatype.java

```diff
@@ -1,19 +1,27 @@
 package org.bahmni.module.bahmnicore.customdatatype.datatype;
 
+import org.bahmni.module.bahmnicore.util.MiscUtils;
 import org.openmrs.Concept;
 import org.openmrs.ConceptAnswer;
 import org.openmrs.api.context.Context;
 import org.openmrs.customdatatype.InvalidCustomValueException;
 import org.openmrs.customdatatype.SerializingCustomDatatype;
 
 import java.util.Collection;
+import java.util.List;
 
 public class CodedConceptDatatype extends SerializingCustomDatatype<Concept> {
     private Concept codedConcept;
 
     @Override
     public void setConfiguration(String id) {
-        this.codedConcept = Context.getConceptService().getConcept(Integer.valueOf(id));
+        if (MiscUtils.onlyDigits(id)) {
+            this.codedConcept = Context.getConceptService().getConcept(Integer.valueOf(id));
+        } else {
+            List<List<Object>> conceptId = Context.getAdministrationService()
+                    .executeSQL("select concept_id from concept where uuid='" + id + "';", true);
```

Comment:
Is this pure SQL or could it be HQL instead?

Reply:
Or, couldn't we use the `ConceptService`?

Reply:
@rbuisson For some unapparent reason the `conceptService` call by `uuid`and `name` throws an exception  ..Only the call by `conceptId` works.

Reply:
Considering time constraints, we will merge this fix and come back to it later.